triage/README.md

```diff
@@ -1,7 +1,161 @@
+      test's failures remaining separate within the global cluster. The mapping at this point is
+      `Global Cluster Text => Test Name => Group of Test Failures`.
+   1. Transform the global clustering into a format that compresses better, and which is consumable
+      by the web page.
+   1. If a mapping of owners to owner prefixes (such as `sig-testing => [sig-testing]`) was provided
+      as a flag, load it and annotate each cluster with an owner.
+   1. Write the results to a JSON file.
+   1. If the `output_slices` flag is set, create individual files ("slices") for each owner (if the
+      owners were provided). Also, split the results into 256 slices based on the cluster IDs.
+   1. Write the slices to JSON files.
+1. Upload the results into Google Cloud Storage so they can be browsed via the web page.
+
+
+## File Structure
+
+Below are the file structures for the ingested and outputted files. `...` denotes a repetition of the
+previous element.
+
+### `builds` Flag
+```json
+[
+   {
+      "path": string,
+      "started": int as string,
+      "elapsed": int as string,
+      "tests_run": int as string,
+      "tests_failed": int as string,
+      "result": string,
+      "executor": string,
+      "job": string,
+      "number": int as string,
+      "pr": string,
+      "key": string
+   },
+   ...
+]
+```
+
+### `tests` Flag
+This is a newline-delimited list of JSON objects. **Note the lack of trailing comma between objects!**
+```json
+{
+   "started": int as string,
+   "build": string,
+   "name": string,
+   "failure_text": string
+}
+...
+```
+
+### `previous` Flag (same as Main Output)
+```json
+{
+   "clustered": [
+      {
+         "key": string,
+         "id": string,
+         "text": string,
+         "spans": [
+            int,
+            ...
+         ],
+         "tests": [
+            {
+               "name": string,
+               "jobs": [
+                  {
+                     "name": string,
+                     "builds": [
+                        int,
+                        ...
+                     ]
+                  },
+                  ...
+               ]
+            },
+            ...
+         ],
+         "owner": string,
+      },
+      ...
+   ],
+   "builds": {
+      "jobs": {
+         string: ([int, ...] OR {int as string: int, ...})  // See the description of the jobCollection type
+      },
+      "cols": {
+         "started": [int, ...],
+         "tests_failed": [int, ...],
+         "elapsed": [int, ...],
+         "tests_run": [int, ...],
+         "result": [string, ...],
+         "executor": [string, ...],
+         "pr": [string, ...]
+      },
+      "job_paths": {
+         string: string,
+         ...
+      },
+   }
+}
+```
+
+### `owners` Flag
+```json
+{
+   string: [
+      string,
+      ...
+   ],
+   ...
+}
+```
+
+### Main Output
+See `previous` flag.
```

Comment:
I think it would make more sense if main output came first, and previous referred to main

Reply:
Fixed